pkg/util/kubernetes/apiserver/apiserver.go

```diff
@@ -98,22 +110,20 @@ func getClient() (*corev1.CoreV1Client, error) {
 		}
 	}
 	k8sConfig.Timeout = 2 * time.Second
-	coreClient, err := corev1.NewForConfig(k8sConfig)
-	return coreClient, err
+
+	return k8sConfig, nil
 }
 
 func (c *APIClient) connect() error {
 	var err error
-	if c.Client == nil {
-		c.Client, err = getClient()
-		if err != nil {
-			log.Errorf("Not Able to set up a client for the Leader Election: %s", err)
-			return err
-		}
+	c.Cl, err = getClientSet()
+	if err != nil {
+		// We do not return an error as the HPA is an option that should not prevent the DCA to work.
+		log.Errorf("Not able to set up a client for the API Server: %s", err)
```

Comment:
I'm not sure this comment makes sense here anymore.

Reply:
We can have an error if `kubernetes.NewForConfig(k8sConfig)` fails. So I think it's still relevant ?